packaging/irods_prc_tests/test_rule_engine_plugin_metadata_guard_atomic.py

```diff
@@ -0,0 +1,57 @@
+from irods.session import iRODSSession
+from irods.meta import iRODSMeta
+from irods.meta import iRODSMeta, AVUOperation
+
+import irods.exception
+
+import sys
+import unittest
+import tempfile
+import shutil
+import os
+import json
+
+class test_metadata_guard(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        with iRODSSession(host='localhost', port=1247, user='rods', password='rods', zone='tempZone') as session:
+            a = session.users.create_with_password('alice', 'test') 
+            with iRODSSession(host='localhost', port=1247, user='alice', password='test', zone='tempZone') as new_session:
+                new_session.data_objects.create("/tempZone/home/alice/test1")
+            coll = session.collections.get("/tempZone")
+            if 'irods::metadata_guard' not in coll.metadata:
+                coll.metadata.add('irods::metadata_guard', '{"prefixes": ["irods::"], "admin_only": true}') 
+            cls.temp_path = os.path.join(tempfile.mkdtemp(), 'irods_config_backup')
+            shutil.copy2('/etc/irods/server_config.json', cls.temp_path)
+            with open('/etc/irods/server_config.json', 'r+') as config_file:
+                config_json = json.load(config_file)
+                config_json['plugin_configuration']['rule_engines'].insert(0, json.loads('{"instance_name":"irods_rule_engine_plugin-metadata_guard-instance","plugin_name":"irods_rule_engine_plugin-metadata_guard","plugin_specific_configuration":{}}'))
+            with open('/etc/irods/server_config.json', 'wt') as config_file:
+                json.dump(config_json, config_file)
```

Comment:
Consider building some utility functions for testing. Depending on how things go, we may port them to other projects or provide a module that makes them available to other projects.